library/lisparray.lisp

```diff
@@ -0,0 +1,81 @@
+;;;; lisparray.lisp
+;;;;
+;;;; An interface to Common Lisp rank-1 SIMPLE-ARRAYs.
+
+(coalton-library/utils:defstdlib-package #:coalton-library/lisparray
+  (:use #:coalton)
+  (:local-nicknames (#:ty  #:coalton-library/types))
+  (:export
+   #:LispArray
+   #:make
+   #:make-uninitialized
+   #:length
+   #:aref
+   #:set!))
+
+(in-package #:coalton-library/lisparray)
+
+(named-readtables:in-readtable coalton:coalton)
+
+#+coalton-release
+(cl:declaim #.coalton-impl/settings:*coalton-optimize-library*)
+
+(coalton-toplevel
+  ;; The representation of (LispArray :t) is specially dealt with by
+  ;; the compiler in lisp-type.lisp.
+  (define-type (LispArray :t)
+    "A one-dimensional, non-resizable array of elements.
+
+These arrays are represented as possibly specialized `(cl:simple-array <type> (cl:*))` and are meant to be used as a tool either to interface with Lisp code or to implement efficient data structures. One should consult `Vector` or `Seq` for more general sequential data structure needs.
+
+Whether or not the arrays are specialized depends on the underlying Lisp implementation. Consult `cl:upgraded-array-element-type` to determine whether `LispArray` may get specialized.")
+
+)                                       ; COALTON-TOPLEVEL
```

Comment:
Are these toplevels split intentionally? If so, then I believe a comment is warranted

Reply:
no, but it would be nice if we discussed the style of thus stuff